cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -88,7 +91,13 @@ func (kc *kubeletConfig) Marshal() ([]byte, error) {
 }
 
 func (kc *kubeletConfig) Unmarshal(docmap kubeadmapi.DocumentMap) error {
-	return kubeletHandler.Unmarshal(docmap, &kc.config)
+	if err := kubeletHandler.Unmarshal(docmap, &kc.config); err != nil {
+		return err
+	}
+
+	// If we have successfully unmarshalled something, then this is no longer a generated config
+	kc.userSupplied = true
```

Comment:
Considering this is implicit in the function, I think the function name should be changed in `UnsmarshalUserSuppliedConfig` or something similar